databricks_cli/permissions/cli.py

```diff
@@ -0,0 +1,188 @@
+JSON_FILE_OPTIONS = ['json-file', 'json']
+CREATE_USER_OPTIONS = JSON_FILE_OPTIONS + ['user-name']
+
+PERMISSIONS_OPTION = ['group-name', 'user-name', 'service-name']
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Get permissions for an item.  ' +
+                          'Possible object types are: \n\t{}\n'
+               .format(PermissionTargets.help_values())
+               )
+@click.argument('object-type')
+@click.argument('object-id', required=False)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_cli(api_client, object_type, object_id):
+    perms_api = PermissionsApi(api_client)
+    if not object_type:
+        click.echo('Possible object types are: {}'.format(PermissionTargets))
+    click.echo(pretty_format(perms_api.get_permissions(object_type, object_id)))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='List permission types')
+@click.argument('object_type')
+@click.argument('object_id')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_permissions_types_cli(api_client, object_type, object_id):
+    perms_api = PermissionsApi(api_client)
+    click.echo(pretty_format(perms_api.get_possible_permissions(object_type, object_id)))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Add or modify permission types')
+@click.argument('object_type')
+@click.argument('object_id')
+@click.option('--group-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--user-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--service-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--permission-level', metavar='<string>', default=None,
+              required=True)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+# pylint:disable=unused-argument
+def add_cli(api_client, object_type, object_id, user_name, group_name, service_name,
+            permission_level):
+    perms_api = PermissionsApi(api_client)
+
+    if not user_name and not group_name and not service_name:
+        click.echo('Need --user-name, --service-name or --group-name')
+        return
+
+    if not permission_level:
+        click.echo('Need permission-level: {}'.format([e.value for e in PermissionLevel]))
+
+    # build the json for the use
+    perm_type = None
+    value = None
+    if user_name:
+        perm_type = PermissionType.user
+        value = user_name
+    elif group_name:
+        perm_type = PermissionType.group
+        value = group_name
+    elif service_name:
+        perm_type = PermissionType.service
+        value = service_name
+    else:
+        click.echo(
+            'Invalid permission-level must be one of {}'.format([e.value for e in PermissionLevel]))
```

Comment:
`nit`: This is PermissionType and not PermissionLevel.

Reply:
Ty, fixed.